devtools/maven/src/main/java/io/quarkus/maven/CodeGenMojo.java

```diff
@@ -0,0 +1,152 @@
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.util.Collections;
+import java.util.List;
+import java.util.Properties;
+import java.util.function.Consumer;
+
+import org.apache.maven.artifact.Artifact;
+import org.apache.maven.plugin.AbstractMojo;
+import org.apache.maven.plugin.MojoExecutionException;
+import org.apache.maven.plugin.MojoFailureException;
+import org.apache.maven.plugins.annotations.Component;
+import org.apache.maven.plugins.annotations.LifecyclePhase;
+import org.apache.maven.plugins.annotations.Mojo;
+import org.apache.maven.plugins.annotations.Parameter;
+import org.apache.maven.plugins.annotations.ResolutionScope;
+import org.apache.maven.project.MavenProject;
+import org.eclipse.aether.RepositorySystem;
+import org.eclipse.aether.RepositorySystemSession;
+import org.eclipse.aether.repository.RemoteRepository;
+
+import io.quarkus.bootstrap.app.CuratedApplication;
+import io.quarkus.bootstrap.app.QuarkusBootstrap;
+import io.quarkus.bootstrap.classloading.QuarkusClassLoader;
+import io.quarkus.bootstrap.model.AppArtifact;
+import io.quarkus.bootstrap.prebuild.CodeGenFailureException;
+import io.quarkus.bootstrap.resolver.maven.MavenArtifactResolver;
+import io.quarkus.deployment.CodeGenerator;
+import io.quarkus.deployment.codegen.CodeGenData;
+
+@Mojo(name = "prepare", defaultPhase = LifecyclePhase.GENERATE_SOURCES, requiresDependencyResolution = ResolutionScope.COMPILE_PLUS_RUNTIME)
+public class CodeGenMojo extends AbstractMojo {
+
+    @Parameter(defaultValue = "${project.build.directory}")
+    private File buildDir;
+
+    @Parameter(defaultValue = "${project}", readonly = true, required = true)
+    protected MavenProject project;
+
+    @Component
+    private RepositorySystem repoSystem;
+
+    @Parameter(defaultValue = "${repositorySystemSession}", readonly = true)
+    private RepositorySystemSession repoSession;
+
+    @Parameter(defaultValue = "${project.remoteProjectRepositories}", readonly = true, required = true)
+    private List<RemoteRepository> repos;
+
+    /** Skip the execution of this mojo */
+    @Parameter(defaultValue = "false", property = "quarkus.prepare.skip")
+    private boolean skipSourceGeneration = false;
+
+    @Override
+    public void execute() throws MojoExecutionException, MojoFailureException {
+        String projectDir = project.getBasedir().getAbsolutePath();
+        Path sourcesDir = Paths.get(projectDir, "src", "main");
+        doExecute(sourcesDir, path -> project.addCompileSourceRoot(path.toString()), false);
+    }
+
+    void doExecute(Path sourcesDir,
+            Consumer<Path> sourceRegistrar,
+            boolean test) throws MojoFailureException, MojoExecutionException {
+        if (project.getPackaging().equals("pom")) {
+            getLog().info("Type of the artifact is POM, skipping build goal");
+            return;
+        }
+        if (skipSourceGeneration) {
+            getLog().info("Skipping Quarkus build");
```

Comment:
Should we be more explicit in this message? We only skip the code gen part right?

Reply:
yes, I'll fix this